### DISCUSSION TITLE
v6.2.0

### DISCUSSION THREAD

**Discussion by snipe** [snipe/snipe-it#13648] [General]
## ⚠️ IMPORTANT: Future versions of Snipe-IT will require PHP8.1 or greater

Welcome to the release notes for Snipe-IT 6.2. We've been trying to fix as many bugs (and sneak in as many small features) as we can before Snipe-IT v7 comes out.

 __Reminder that Snipe-IT v7 WILL REQUIRE PHP 8.1 or higher!! This is a slightly higher version than the PHP 8.0 version we've been saying for a while now, so please make sure you're ready.__ The reason for this slight bump in minimum requirements is that we've decided to launch Snipe-IT v7 on Laravel 10, which requires that slightly higher version. Our goal is to have Snipe-IT v7 out in a month or two. 

## Highlights

This release includes lots of bug fixes detailed below. It introduces a new label engine which will hopefully be a big help to those folks who have struggled with getting their specific printers to work with Snipe-IT labels, but also has a legacy fallback, so if you have things just right, nothing needs to change for now. 

We've also added back the sticky header for long lists, and column-reordering in those table views:

https://github.com/snipe/snipe-it/assets/197404/6b39d2af-4d06-4aca-9efd-5acb242eb4e9

Additionally, we made some improvements to the importer, added a minimum quantity on models so you will get those alerts in your bell icon when you don't have enough of a particular asset model available. And we've fixed a few small inconsistencies in the API with regards to boolean fields. 

__Important!__ SCIM support did not previously support changing email addresses via Azure and some other SCIM clients. Under v6.2, that has been fixed, but if you were depending on the previous behavior, you may get unexpected results.

## What's Changed
* Change settings table `saml_idp_metadata` to MEDIUMTEXT by @snipe in https://github.com/snipe/snipe-it/pull/13255
* Check for max PHP version in upgrader by @snipe in https://github.com/snipe/snipe-it/pull/13257
* Making Serial Number the logic checked against for new imports by @akemidx in https://github.com/snipe/snipe-it/pull/13169
* Fix label wrapping on category edit page by @marcusmoore in https://github.com/snipe/snipe-it/pull/13264
* Have ChipperCi run on each PR by @marcusmoore in https://github.com/snipe/snipe-it/pull/13265
* Applied Snyk #13272 by @snipe in https://github.com/snipe/snipe-it/pull/13278
* Upgraded acorn by @snipe in https://github.com/snipe/snipe-it/pull/13279
* [Snyk] Upgrade webpack from 5.86.0 to 5.87.0 #13267 by @snipe in https://github.com/snipe/snipe-it/pull/13280
* Fixes dev dependency on PHP 7.4 by @marcusmoore in https://github.com/snipe/snipe-it/pull/13282
* Fixed #13276 Error Showing Requested Assets by @inietov in https://github.com/snipe/snipe-it/pull/13281
* Added phone, fax to departments, locations, companies by @snipe in https://github.com/snipe/snipe-it/pull/13277
* Run ChipperCI on all PRs (take three) by @marcusmoore in https://github.com/snipe/snipe-it/pull/13293
* Avoid accidentally wiping local database when running tests by @marcusmoore in https://github.com/snipe/snipe-it/pull/13292
* Fixed License importer missing Checkout to: Username and Checkout to: Email in dropdown [sc-23456] by @inietov in https://github.com/snipe/snipe-it/pull/13268
* fixed margin if `qr_code present` by @Godmartinz in https://github.com/snipe/snipe-it/pull/13233
* Fixed #13167 Default location not being set/updated upon check-in by @inietov in https://github.com/snipe/snipe-it/pull/13225
* Fixed #13256 - Added option to switch to localStorage instead of cookies by @snipe in https://github.com/snipe/snipe-it/pull/13302
* Better auto increment asset-tags [FD-32892] by @uberbrady in https://github.com/snipe/snipe-it/pull/13301
* Added Inline markdown method by @snipe in https://github.com/snipe/snipe-it/pull/13288
* User department now visible in side pane of asset view page by @akemidx in https://github.com/snipe/snipe-it/pull/13184
* Fixed: deleted users showing in current users by @snipe in https://github.com/snipe/snipe-it/pull/13311
* Added ability to show/hide custom fields in list views by default by @snipe in https://github.com/snipe/snipe-it/pull/13312
* Moved warranty into manufactuer’s section by @snipe in https://github.com/snipe/snipe-it/pull/13315
* Fixed Employee Number no longer importing [sc-23497] by @inietov in https://github.com/snipe/snipe-it/pull/13318
* Fixes #13251 - ldap_location attribute by @uberbrady in https://github.com/snipe/snipe-it/pull/13322
* Fixed the inability to load location select when editing profile by @marcusmoore in https://github.com/snipe/snipe-it/pull/13310
* Added test cases around user search by @marcusmoore in https://github.com/snipe/snipe-it/pull/13319
* Added strtolower for category type by @snipe in https://github.com/snipe/snipe-it/pull/13333
* Update league/oauth2-server for CVE-2023-3726  by @joelpittet in https://github.com/snipe/snipe-it/pull/13299
* Current value added to asset index page by @akemidx in https://github.com/snipe/snipe-it/pull/13197
* Fixes #13341: Clean up correct seats on license delete by @Loemnk in https://github.com/snipe/snipe-it/pull/13342
* Fixes bug/sc-23509: Added manufacturer back into license importer by @snipe in https://github.com/snipe/snipe-it/pull/13344
* Check that `$this->get_depreciation()` returns by @snipe in https://github.com/snipe/snipe-it/pull/13346
* Fixed: tighter check for valid model by @snipe in https://github.com/snipe/snipe-it/pull/13349
* Possible alternative to fixing #13296 - custom report failing when th… by @snipe in https://github.com/snipe/snipe-it/pull/13347
* Delete button on asset view page by @akemidx in https://github.com/snipe/snipe-it/pull/12940
* Fixed #13298 "No records Found" when Switching between Single and Multi-Page Licenses by @inietov in https://github.com/snipe/snipe-it/pull/13363
* Fixed custom fieldset order  by @Godmartinz in https://github.com/snipe/snipe-it/pull/13361
* Fixed #13376 Component checkout via API returns error by @inietov in https://github.com/snipe/snipe-it/pull/13379
* Fixed #11794 Admins Cannot View Encrypted Field by @inietov in https://github.com/snipe/snipe-it/pull/13295
* Updated BS table and table export plugins by @snipe in https://github.com/snipe/snipe-it/pull/13386
* Fixed #13300 Custom Asset Report, Checkout date range never includes current day by @inietov in https://github.com/snipe/snipe-it/pull/13309
* Fixed `webhook_endpoint` data type from varchar to a text by @Godmartinz in https://github.com/snipe/snipe-it/pull/13392
* Fixed #13372: Put guard around assigning location via LDAP by @uberbrady in https://github.com/snipe/snipe-it/pull/13397
* Fixed #13417 - Added mailgun endpoint option, defaulting to the US by @snipe in https://github.com/snipe/snipe-it/pull/13418
* Re-add the option for asset_eol_date as an import field by @uberbrady in https://github.com/snipe/snipe-it/pull/13421
* Fixed tooltip not loading on encrypted field lock icon on asset detail view by @snipe in https://github.com/snipe/snipe-it/pull/13424
* Updated webpack by @snipe in https://github.com/snipe/snipe-it/pull/13425
* Added messaging to category edit page to improve clarity around when users will be emailed by @marcusmoore in https://github.com/snipe/snipe-it/pull/13407
* Fixes ChipperCi Pipeline by @marcusmoore in https://github.com/snipe/snipe-it/pull/13414
* Updated autotable by @snipe in https://github.com/snipe/snipe-it/pull/13428
* Removed unused Codeception test code by @marcusmoore in https://github.com/snipe/snipe-it/pull/13432
* Improved test stability and messaging by @marcusmoore in https://github.com/snipe/snipe-it/pull/13434
* Removed unneeded `Company::scopeCompanyables` from `ReportsController` by @marcusmoore in https://github.com/snipe/snipe-it/pull/13431
* Fixed route for accessory index by @snipe in https://github.com/snipe/snipe-it/pull/13439
* Fixed case in BS table formatter for boolean by @snipe in https://github.com/snipe/snipe-it/pull/13445
* Fixed #13450 - remove escaping on saveing new custom fields by @snipe in https://github.com/snipe/snipe-it/pull/13454
* Allowed permissions in the user factory to be chained by @marcusmoore in https://github.com/snipe/snipe-it/pull/13437
* Fixed #13446 - added translations to status label type column by @Godmartinz in https://github.com/snipe/snipe-it/pull/13448
* Fixed #13382 - confusion around localization date by @snipe in https://github.com/snipe/snipe-it/pull/13455
* Updated language strings from CrowdIn by @snipe in https://github.com/snipe/snipe-it/pull/13461
* Add `email` to companies by @Robert-Azelis in https://github.com/snipe/snipe-it/pull/13442
* New Label Engine by @cram42 in https://github.com/snipe/snipe-it/pull/12050
* Fixed unclosed brace by @snipe in https://github.com/snipe/snipe-it/pull/13465
* Fixes weird label engine layout by @snipe in https://github.com/snipe/snipe-it/pull/13466
* Accessory checkin via API reported wrong target user by @uberbrady in https://github.com/snipe/snipe-it/pull/13463
* Fixes more label html tweaks by @snipe in https://github.com/snipe/snipe-it/pull/13467
* Fixed checkboxes for accessibility by @snipe in https://github.com/snipe/snipe-it/pull/13469
* Fixed #13464 typo in accessories clone by @snipe in https://github.com/snipe/snipe-it/pull/13470
* Fixed #13365 - Added `LOGIN_AUTOCOMPLETE` as env var by @snipe in https://github.com/snipe/snipe-it/pull/13457
* Check that there is a company before trying to get name property by @snipe in https://github.com/snipe/snipe-it/pull/13473
* Accessory checkin via API reported wrong target user by @uberbrady in https://github.com/snipe/snipe-it/pull/13476
* Fixed #13409 Search in component detail view not working by @inietov in https://github.com/snipe/snipe-it/pull/13471
* Fixed #13359 Advanced search by Default Location does not work by @inietov in https://github.com/snipe/snipe-it/pull/13483
* Fixed Ldap sync field clear bug by @Godmartinz in https://github.com/snipe/snipe-it/pull/13391
* Fixed #13487 - include supplier url in listing by @snipe in https://github.com/snipe/snipe-it/pull/13488
* Make sure the columns and rows can never be 0 by @snipe in https://github.com/snipe/snipe-it/pull/13489
* Removed the black label from around the default labels in new label engine by @snipe in https://github.com/snipe/snipe-it/pull/13491
* Added Dymo labelwriter by @snipe in https://github.com/snipe/snipe-it/pull/13492
* Fixed passing invalid argument to `strpos()` by @marcusmoore in https://github.com/snipe/snipe-it/pull/13456
* Bumped dependencies by @marcusmoore in https://github.com/snipe/snipe-it/pull/13436
* Added better handling of information of asset history  by @Godmartinz in https://github.com/snipe/snipe-it/pull/13485
* Bulk Editing Custom Fields by @spencerrlongg in https://github.com/snipe/snipe-it/pull/12761
* Fixed #13317 Accessories declined by user remain assigned by @inietov in https://github.com/snipe/snipe-it/pull/13453
* Fixed #13495 added setting for name order by @snipe in https://github.com/snipe/snipe-it/pull/13496
* Small fixes to checkbox layout in general settings blade by @snipe in https://github.com/snipe/snipe-it/pull/13501
* Fixed Asset acceptance error when user company and asset company don't match by @inietov in https://github.com/snipe/snipe-it/pull/13400
* Fixed asset model query in action log transformer by @marcusmoore in https://github.com/snipe/snipe-it/pull/13509
* Fixed #13510 - do not use translations in migrations by @snipe in https://github.com/snipe/snipe-it/pull/13511
* Fixed undeclared variable in ActionlogsTransformer by @marcusmoore in https://github.com/snipe/snipe-it/pull/13525
* Clearer labels barcode text by @snipe in https://github.com/snipe/snipe-it/pull/13533
* Fixed #13521 - make modal “select files” button wider by @snipe in https://github.com/snipe/snipe-it/pull/13535
* Fixed #13516 - Use int not string if no DB_PORT specified by @snipe in https://github.com/snipe/snipe-it/pull/13538
* Run tests for PHP 7.4, 8.0, and 8.1.1 via GitHub Actions by @marcusmoore in https://github.com/snipe/snipe-it/pull/13518
* Fixed #13500 - Try to prevent the browser from pre-filling the LDAP password by @snipe in https://github.com/snipe/snipe-it/pull/13537
* Fixed #12548 Cloning a model blanks fieldset of new one by @inietov in https://github.com/snipe/snipe-it/pull/13528
* Fixed user search not adhering to company scoping by @marcusmoore in https://github.com/snipe/snipe-it/pull/13544
* Don’t cast as boolean, do validate as boolean for User validation  by @snipe in https://github.com/snipe/snipe-it/pull/13546
* Added eager loading for `changedinfo` and removed encrypted values from Asset history by @Godmartinz in https://github.com/snipe/snipe-it/pull/13507
* Removed Dusk by @marcusmoore in https://github.com/snipe/snipe-it/pull/13550
* Added model name and number to dynamic url by @snipe in https://github.com/snipe/snipe-it/pull/13556
* Added last check in column and filter to custom asset report by @marcusmoore in https://github.com/snipe/snipe-it/pull/13498
* Added users total cost of assets to user profile by @Godmartinz in https://github.com/snipe/snipe-it/pull/12901
* Account for hard-deleted models, suppliers in asset history by @snipe in https://github.com/snipe/snipe-it/pull/13565
* Added/Fixed: track changes on asset checkin/out by @johnson-yi in https://github.com/snipe/snipe-it/pull/13557
* Fixed ErrorException: Attempt to read property `asset_tag` on null (rollbar #3541) by @inietov in https://github.com/snipe/snipe-it/pull/13547
* Add default location to tracked changes on checkin actionlog by @johnson-yi in https://github.com/snipe/snipe-it/pull/13567
* Updated testing documentation by @marcusmoore in https://github.com/snipe/snipe-it/pull/13548
* Fixed Attempt to read property require_acceptance on null [rollbar-3557] by @inietov in https://github.com/snipe/snipe-it/pull/13571
* Upgrade to new branch for our fork of laravel-scim-server by @uberbrady in https://github.com/snipe/snipe-it/pull/13573
* Fixed Unaccepted Assets report has incorrect people [freshdesk-37808] by @inietov in https://github.com/snipe/snipe-it/pull/13574
* Fixed #13562 - Added inline file link by @snipe in https://github.com/snipe/snipe-it/pull/13566
* Added validation around `department_id` in API patch request by @marcusmoore in https://github.com/snipe/snipe-it/pull/13575
* Fixed failing test by @marcusmoore in https://github.com/snipe/snipe-it/pull/13577
* Added threshold notifications and min qty for Asset models by @Godmartinz in https://github.com/snipe/snipe-it/pull/13526
* Fixed Call to a member function checkedOutToUser() on null [rollbar-3598] by @inietov in https://github.com/snipe/snipe-it/pull/13586
* Fixed nullability on asset model `min qty` column by @Godmartinz in https://github.com/snipe/snipe-it/pull/13595
* Reset checkout to value when navigating away from type  by @marcusmoore in https://github.com/snipe/snipe-it/pull/13597
* Upgraded actions/checkout to v4 (via #13580) by @snipe in https://github.com/snipe/snipe-it/pull/13581
* Fixed ErrorException: Attempt to read property "id" on null (rollbar #3541) by @inietov in https://github.com/snipe/snipe-it/pull/13549
* Fixed Not reassignable Licenses shouldn't show 'Checkin All Seats' button [sc-23506] by @inietov in https://github.com/snipe/snipe-it/pull/13520
* Added `CompanyableTrait` to `Department` model by @marcusmoore in https://github.com/snipe/snipe-it/pull/13415
* Fixed #13336 - Save unhashed password if no password provided by @snipe in https://github.com/snipe/snipe-it/pull/13343
* Fixed checkbox formatting on quickstart by @snipe in https://github.com/snipe/snipe-it/pull/13602
* Removed ChipperCI integration by @marcusmoore in https://github.com/snipe/snipe-it/pull/13604
* Simplify upload messaging, handle -1% error by @snipe in https://github.com/snipe/snipe-it/pull/13605
* Fixed #13592 - unable to delete model from view page by @snipe in https://github.com/snipe/snipe-it/pull/13606
* Use relative path in backups for cleaner directory structure by @snipe in https://github.com/snipe/snipe-it/pull/13608
* Fixed footer sizing issue by @Godmartinz in https://github.com/snipe/snipe-it/pull/12869
* Fixes #13609 - disallow sorting on age by @snipe in https://github.com/snipe/snipe-it/pull/13610
* Fixed regression that disabled sticky headers by @snipe in https://github.com/snipe/snipe-it/pull/13612
* Added column re-ordering by @snipe in https://github.com/snipe/snipe-it/pull/13613
* Replaces #13618 - Bump docker/build-push-action from 4 to 5 by @snipe in https://github.com/snipe/snipe-it/pull/13625
* Replaces #13617 - Bump docker/setup-buildx-action from 2 to 3 by @snipe in https://github.com/snipe/snipe-it/pull/13626
* Replaces #13616 - Bump docker/metadata-action from 4 to 5 by @snipe in https://github.com/snipe/snipe-it/pull/13627
* Made develop the target branch for dependabot by @snipe in https://github.com/snipe/snipe-it/pull/13623
* Fixed #13628 - removed duplicate favicon tag by @snipe in https://github.com/snipe/snipe-it/pull/13629
* Added unique() to some factory properties by @marcusmoore in https://github.com/snipe/snipe-it/pull/13635
* Added factories to autolabeler action by @marcusmoore in https://github.com/snipe/snipe-it/pull/13634
* Adds tag for docker builds by @spencerrlongg in https://github.com/snipe/snipe-it/pull/12992
* Added Khmer, updated translations by @snipe in https://github.com/snipe/snipe-it/pull/13647

## New Contributors
* @Loemnk made their first contribution in https://github.com/snipe/snipe-it/pull/13342
* @cram42 made their first contribution in https://github.com/snipe/snipe-it/pull/12050

**Full Changelog**: https://github.com/snipe/snipe-it/compare/v6.1.2...v6.2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/snipe/snipe-it/releases/tag/v6.2.0'>v6.2.0</a>.</em>

**Comment by yetanothersys-admin** [comment]
Great update! Thank you all! 😍

Question about the new label engine tho - is there a way to add custom labels to those that already exist as "templates" @cram42 ? We're probably gonna use the "old" engine for the labels still since I didn't see a way to configure a template / make a new one.

But all in all great update! ❤️

**Reply by snipe** [reply]
Making a new template is a bit of work - we're happy to work with you on sizing, etc, but there isn't currently a way to do that without touching code. You can see the isolated Dymo PR here which might provide some guidance: https://github.com/snipe/snipe-it/pull/13492

**Reply by yetanothersys-admin** [reply]
Alright, the old label still works like a charm so not a problem really :D
But thx for the response!
